File: app/src/main/java/hu/bme/szn/randomjokes/ui/editjokes/EditJokesPresenter.java
package hu.bme.szn.randomjokes.ui.editjokes;

import javax.inject.Inject;

import hu.bme.szn.randomjokes.ui.Presenter;
import hu.bme.szn.randomjokes.model.Joke;

public class EditJokesPresenter extends Presenter<EditJokesScreen> {

    @Inject
    public EditJokesPresenter() {
    }

    @Override
    public void attachScreen(EditJokesScreen screen) {
        super.attachScreen(screen);
    }

    @Override
    public void detachScreen() {
        super.detachScreen();
    }


    public void editJoke(
            Long id, Joke joke) {
        screen.editJoke(id, joke);
    }
}

File: app/src/main/java/hu/bme/szn/randomjokes/ui/editjokes/EditJokesPresenter.kt
package hu.bme.szn.randomjokes.ui.editjokes

import hu.bme.szn.randomjokes.model.Joke
import hu.bme.szn.randomjokes.ui.Presenter

class EditJokesPresenter : Presenter<EditJokesScreen>() {

    override fun attachScreen(screen: EditJokesScreen) {
        super.attachScreen(screen)
        //TODO
    }

    override fun detachScreen() {
        super.detachScreen()
        //TODO
    }

    fun editJoke(joke: Joke) {
        screen?.editJoke(joke)
    }

}